actions/online-sign/action.yml

```diff
@@ -67,4 +68,7 @@ runs:
             repo: context.repo.repo,
             workflow_id: 'publish.yml',
             ref: 'publish',
+            inputs: {
+              ref: '${{ steps.move-publish-branch.outputs.rev }}',
+            },
```

Comment:
if we give the actual ref as argument to createWorkflowDispatch (instead of using "publish") would that work without the added action args? Or am I missing something?

Reply:
I may have to read the whole publishing code again... I'm not quite sure how it chooses the correct commit right now :joy:

Reply:
Yeah, it can work! I have to look at how the official checkout action works! Good idea.

Reply:
According to the doc, I don't think it would work. The `ref` has to be a branch name of tag: 
> The git reference for the workflow. The reference can be a branch or tag name.

Reply:
And it kind of makes sense, as the action still makes a checkout, and so would use the tip of the provided branch.

Reply:
so my understanding/assumption:
* actions/checkout defaults to the ref of the event that triggers the workflow (I assume this holds even though the checkout happens as part of a composite action)
* so as long as the publish workflow is triggered with the ref we want, this all should work without additional inputs
* I assume we can just use hash ref as the ref argument to createWorkflowDispatch -- I don't think the docs are making the case that only branch or tag would work?

This should be simple to test...

Reply:
> (I assume this holds even though the checkout happens as part of a composite action) 

Yes, if `ref` is empty (i.e not set which is the case today) it defaults to the ref that triggered the event, i.e `publish`.

> so as long as the publish workflow is triggered with the ref we want, this all should work without additional inputs

Yes, according to the doc, only branch or tag is allowed. but let me test that, I have a repo ready for that.

Reply:
Also tried with building a ref from the commit:

>  message: 'No ref found for: git/commits/8a323a9f7132350da18c54f87f8623f7c021d680',

Reply:
So the documentation seems to be correct. When dispatching a workflow, you need to provide a tag or a branch name.

Reply:
got it, thanks for testing